dsls/sql/src/main/java/org/apache/beam/dsls/sql/example/BeamSqlExample.java

```diff
@@ -17,93 +17,62 @@
-import org.apache.calcite.rel.type.RelProtoDataType;
 import org.apache.calcite.sql.type.SqlTypeName;
-import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
- * This is one quick example.
+ * This is a quick example, which uses Beam SQL DSL to create a data pipeline.
  *
- * <p>Before start, follow https://kafka.apache.org/quickstart to setup a Kafka
- * cluster locally, and run below commands to create required Kafka topics:
- * <pre>
- * <code>
- * bin/kafka-topics.sh --create --zookeeper localhost:2181 --replication-factor 1 \
- *   --partitions 1 --topic orders
- * bin/kafka-topics.sh --create --zookeeper localhost:2181 --replication-factor 1 \
- *   --partitions 1 --topic sub_orders
- * </code>
- * </pre>
- * After run the application, produce several test records:
- * <pre>
- * <code>
- * bin/kafka-console-producer.sh --broker-list localhost:9092 --topic orders
- * invalid,record
- * 123445,0,100,3413423
- * 234123,3,232,3451231234
- * 234234,0,5,1234123
- * 345234,0,345234.345,3423
- * </code>
- * </pre>
- * Meanwhile, open another console to see the output:
- * <pre>
- * <code>
- * bin/kafka-console-consumer.sh --bootstrap-server localhost:9092 --topic sub_orders
- * **Expected :
- * 123445,0,100.0
- * 345234,0,345234.345
- * </code>
- * </pre>
  */
-public class BeamSqlExample implements Serializable {
+public class BeamSqlExample {
+  private static final Logger LOG = LoggerFactory.getLogger(BeamSqlExample.class);
 
   public static void main(String[] args) throws Exception {
-    PipelineOptions options = PipelineOptionsFactory.fromArgs(new String[] {}).withValidation()
-        .as(PipelineOptions.class); // FlinkPipelineOptions.class
-    options.setJobName("BeamSqlExample");
-    Pipeline pipeline = Pipeline.create(options);
+    PipelineOptions options = PipelineOptionsFactory.create();
+    Pipeline p = Pipeline.create(options);
 
-    BeamSQLEnvironment runner = BeamSQLEnvironment.create();
-    runner.addTableMetadata("ORDER_DETAILS", getTable("127.0.0.1:9092", "orders"));
-    runner.addTableMetadata("SUB_ORDER", getTable("127.0.0.1:9092", "sub_orders"));
+    //define the input row format
+    BeamSQLRecordType type = new BeamSQLRecordType();
+    type.addField("c1", SqlTypeName.INTEGER);
+    type.addField("c2", SqlTypeName.VARCHAR);
+    type.addField("c3", SqlTypeName.DOUBLE);
+    BeamSQLRow row = new BeamSQLRow(type);
+    row.addField(0, 1);
+    row.addField(1, "row");
+    row.addField(2, 1.0);
 
-    // case 2: insert into <table>(<fields>) select STREAM <fields> from
-    // <table> from <clause>
-    String sql = "INSERT INTO SUB_ORDER(order_id, site_id, price) " + "SELECT "
-        + " order_id, site_id, price " + "FROM ORDER_DETAILS " + "WHERE SITE_ID = 0 and price > 20";
+    //create a source PCollection with Create.of();
+    PCollection<BeamSQLRow> inputTable = PBegin.in(p).apply(Create.of(row));
```

Comment:
You should use `Create.of(row).withCoder(new BeamSqlRowCoder(type))`, it removes the need to call `inputTable.setCoder()`. This allows Create to encode/decode the records using the supplied coder instead of using Java serialization.